### ISSUE TITLE
Rumble bug in some emulators

### ISSUE THREAD

**Issue by chrisclue** [Struggleton/Delfinovin#10]
I got this bug which the controller kept rumbling after receiving consecutive rumbling.
The only way to stop the controller of rumbling is getting to rumble again with some actions on the game, or disconnecting the gamecube controller. This only happened to me in Retroarch and CEMU (In dolphin and other games works without issues), and sometimes the vibration isn't strong enough, mostly like it doesn't rumble at all.

I got to say that the same issue of the vibrations keeping rumbling happend to me in CEMU using the API of the Gamecube Controller that the emulator have, but in the recently realese (1.22.12c) seems to be fix it for me.

I'm also use a PS3 controller with the ScpToolkit from Nefarius and these emulator work fine with not problem in the Xinput vibration. I'm using an unofficial and generic Gamecube adapter, I don't know if other gamecube adapters have the same problem.

**Comment by Struggleton** [comment]
Very late response - Is this still happening? If so is there a way to replicate without CEMU / Retroarch? Also just from testing in CEMU's input settings by dragging the slider it seems to work fine (1.25.1c, SDL + emulated Wii U Pro Controller.)

**Comment by chrisclue** [comment]
Its seems in the latest version of CEMU it works already fine. But is still hapenning only in Retroarch, besides that I didn't find in other emulator or game this issue. Also i can replicate this bug using the Dolphin emulator which let you test the rumble and select the motor L, R or both but sometimes didnt' rumble.
I don't know if is related to the fact that the gamecube controller only have one rumble motor besides other controllers that have two.

**Comment by Struggleton** [comment]
I've drafted a new release with the fixes, [v1.1.1](https://github.com/Struggleton/Delfinovin/releases/tag/v1.1.1) and commit https://github.com/Struggleton/Delfinovin/commit/a8464b8a8ccd98425b23dd8ceb8a5c00a4ebe22d. Please test the fixes and let me know if they work for you in Retroarch - they worked in Dolphin from what I could test

**Comment by chrisclue** [comment]
Thanks for the fix. In dolphin the L and R motor works really well separately, but when both motors are selected the bug still happen, also for retroarch.
The configuration of Dolphin looks like this to replicate the bug.
![image](https://user-images.githubusercontent.com/84299120/132087582-ea4939a3-edbc-482a-bfec-9678fadecb91.png)

**Comment by Struggleton** [comment]
That's extremely strange. I can not replicate this using the same setup you've listed here - It seems to work with both motors selected and rumble enabled in Delfinovin.

**Comment by Struggleton** [comment]
For the consecutive rumbling bug, I'm unable to replicate it using the set up you've provided for Dolphin. Is it semi random when it occurs? Similarly after v1.1.1 using the setup you've sent I can't replicate your issue - When setting both motors to rumble it rumbles

**Comment by chrisclue** [comment]
It's very odd indeed . I got to mention that I use the latest dolphin version in the previous configuration
but using Dolphin 5.0 (stable) I notice that using this configuration works great
![image](https://user-images.githubusercontent.com/84299120/132164421-757a83cd-d0d2-4c08-904d-bb9c44de7880.png)
 but when i select the option "| OR" to select the Motor R works fine the first time, but the second time the bug happend or sometimes is random.
![image](https://user-images.githubusercontent.com/84299120/132164989-9a7bdaf8-b844-4657-9b90-912b96275c3d.png)

If that don't replicate the bug then i guess maybe it's something about my generic adapter and the only thing to do will be wait if someone else having this problem.

**Comment by Struggleton** [comment]
Sorry, I'm still unable to replicate it. The rumble does end when both are selected and takes into account the length of both. I'm using a Mayflash adapter and have tested using a Nyko one as well with similar results

**Comment by SamMemory** [comment]
Hello, so I found out about Delfinovin from Parsec tutorial video, I've been patient for so long until somebody figured out on how to get the official Nintendo GameCube Controller Adapter to recognizes as an XInput Device. I gave Delfinovin a try, so far it works phenomenal. Works with both Wired & Wavebird Wireless controller, as it would with Dolphin Native Support. 

I will say the only thing I notice is that when swapping out or unplug my controller with a different controller in game while the app is running, my control stick ranges starts to mess up, my character runs in some direction slower. The only fix is to quit the app & never unplug or switch controllers until the app is close.  

I tinkered with both control stick ranges, so far these settings worked for me:
![Delfinovin Config](https://user-images.githubusercontent.com/90251715/132601070-f6db08b9-f802-436b-b5a2-c76705d1c900.PNG)

I've also had this issue where my controller kept Rumbling, it kinda fixes itself when I perform an action where the rumble plays a roll. It happened on latest Retroarch running current version of Mupen64Plus, tried it with Paper Mario 64, when Mario uses the hammer or ground pounds, my controller keeps vibrating, if I repeats these action, it randomly can stop vibrating (sometime it doesn't always work). When I repeat the action it keeps rumbling again. Hopefully this should somehow replicate the issue. 

Rumble setting on Mupen64Plus (Retroarch)
![Screenshot 2021 09 09 - 01 19 57 45](https://user-images.githubusercontent.com/90251715/132602149-bf79cd75-d966-45bb-9949-da73ab43a66f.png)

**Comment by chrisclue** [comment]
> I've also had this issue where my controller kept Rumbling, it kinda fixes itself when I perform an action where the rumble plays a roll. It happened on latest Retroarch running current version of Mupen64Plus, tried it with Paper Mario 64, when Mario uses the hammer or ground pounds, my controller keeps vibrating, if I repeats these action, it randomly can stop vibrating (sometime it doesn't always work). When I repeat the action it keeps rumbling again. Hopefully this should somehow replicate the issue.
> 
> Rumble setting on Mupen64Plus (Retroarch)
> ![Screenshot 2021 09 09 - 01 19 57 45](https://user-images.githubusercontent.com/90251715/132602149-bf79cd75-d966-45bb-9949-da73ab43a66f.png)

That's happend also for me. Have you been able to replicate the rumble bug using the Dolphin emulator?. 
I was thinking maybe the way that the program send data to the rumble cause the bug, and in Dolphin the option "| OR" is the key to understand that, but i don't know exactly what does that "| OR" thing in difference of selecting both motors normally.

**Comment by chrisclue** [comment]
I found a workaround for the problem with Retroarch, in the steam version of retroarch the rumble works fine.

**Comment by BenPayton** [comment]
I started having this issue while playing the Ship of Harkinian Zelda: Ocarina of Time PC Port. I'm using a genuine GameCube controller with a 3rd party GC controller adapter. I started noticing this when I got to the part of the game where you start using bombs and there's a lot more rumble action on the controller. I'm not 100% sure but it seems to happen when more than one rumble action triggers at the same time. It does eventually stop if I trigger another rumble action like striking the wall with the sword, but obviously it's quite annoying.
![Capture](https://user-images.githubusercontent.com/19674471/207462551-7330f827-a6d6-4b73-99a3-00e1553b5ced.PNG)
![Capture2](https://user-images.githubusercontent.com/19674471/207462868-dc45a4a3-b72a-4271-8312-fc2062bc15b5.PNG)
